Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM --platform=linux/amd64 ruby:3.0.6
+RUN rm /bin/sh && ln -s /bin/bash /bin/sh
+WORKDIR /usr/src/app
+EXPOSE 3000
+
+RUN apt-get update -y && apt-get install -y --no-install-recommends \
+  default-jre \
+  default-mysql-client \
+  imagemagick \
+  libprotobuf-dev \
+  protobuf-compiler 
+
+# Workaround for PhantomJS: https://github.com/DMOJ/online-judge/pull/1270
+ENV OPENSSL_CONF /etc/ssl/
+
+ENV NODE_VERSION 16.18.1
+ENV NVM_DIR=/root/.nvm
+
+RUN apt install -y curl \
+  && curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash \
+  && . "$NVM_DIR/nvm.sh" \
+  && nvm install $NODE_VERSION \
+  && nvm alias default $NODE_VERSION \
+  && nvm use default \
+  && npm install -g yarn \
+  && yarn install
```

Comment:
We need to copy the `package.json` and `yarn.lock` files before running the `yarn install`.

Reply:
implemented